addons/product/models/product_pricelist_item.py

```diff
@@ -477,7 +477,8 @@ def _is_applicable_for(self, product, qty_in_product_uom):
 
         elif self.applied_on == "2_product_category":
             if (
-                product.categ_id != self.categ_id
+                product.categ_id
+                and product.categ_id != self.categ_id
```

Comment:
Shouldn't this first condition be inversed? My understanding is that if there is no category on the product, `res` should be `False` as the pricelist item cannot be applied. So something like:
```suggestion
                if not product.categ_id or (
                    product.categ_id != self.categ_id
                    and not product.categ_id.parent_path.startswith(self.categ_id.parent_path)
                ):
```

Reply:
Hello @vava-odoo ,

I have made the changes.
Please Review this PR.  😊

Thank you